cmd/stats/stats.go

```diff
@@ -185,19 +197,185 @@ func argsToColumnFilter(cmd *cobra.Command) []string {
+		}
+	}
+	avg = sum / float64(len(reps))
+
+	colData := []float64{avg, min, max, first, last, delta}
+
+	row := []string{}
+
+	for _, d := range colData {
+		row = append(row, util.ColumnString(colName, d))
+	}
+	return row
+}
+
+func printStats(cmd *cobra.Command, reps []Rep) {
+	name := cmd.Flag(NAME).Value.String()
+	lang := cmd.Flag(LANGUAGE).Value.String()
+	start := cmd.Flag(START).Value.String()
+	if since := cmd.Flag("since").Value.String(); since != "" {
+		start = since
+	}
+	end := cmd.Flag(END).Value.String()
+
 	// print the header
-	fmt.Printf("%s\n", strings.Join(cols, "\t"))
+	fmt.Println(getStatsHeader(name, lang, start, end))
+
+	cols := argsToColumnFilter(cmd)
+
+	if len(reps) == 0 {
+		fmt.Println("no stats")
+	} else if len(reps) > 1 {
+
+		// print the stats table
+		table := tw.NewWriter(os.Stdout)
+		table.SetHeader([]string{"", "avg", "min", "max", "first", "last", "delta"})
+		table.SetAutoWrapText(false)
+		table.SetAutoFormatHeaders(true)
+		table.SetHeaderAlignment(tw.ALIGN_LEFT)
+		table.SetAlignment(tw.ALIGN_LEFT)
+		table.SetCenterSeparator("")
+		table.SetColumnSeparator("")
+		table.SetRowSeparator("")
+		table.SetHeaderLine(false)
+		table.SetTablePadding("  ") // pad with tabs
+		table.SetNoWhiteSpace(true)
+		for _, col := range cols {
+			if col == START || col == NAME {
+				continue
+			}
+			row := []string{col}
+			row = append(row, getColumnStats(reps, col)...)
+			table.Append(row)
+		}
+		table.Render()
+
+		wpmGraphData := []float64{}
+		rawWpmGraphData := []float64{}
+		mistakesGraphData := []float64{}
+		accuracyGraphData := []float64{}
+		errorsGraphData := []float64{}
+		for i := range reps {
+			// reverse the order of the reps
+			// so the date increases as X increases
+			currRep := reps[len(reps)-1-i]
+			wpmGraphData = append(wpmGraphData, currRep.Wpm)
+			mistakesGraphData = append(mistakesGraphData, float64(currRep.Miss))
+			rawWpmGraphData = append(rawWpmGraphData, currRep.Raw)
+			accuracyGraphData = append(accuracyGraphData, currRep.Acc)
+			errorsGraphData = append(errorsGraphData, float64(currRep.Errs))
+		}
 
+		// the data for each column's graph
+		graphDatum := [][]float64{accuracyGraphData, errorsGraphData, mistakesGraphData, rawWpmGraphData, wpmGraphData}
+		graphColors := []g.AnsiColor{g.Green, g.Red, g.Yellow, g.Gray, g.Default}
+		graphLegends := []string{ACCURACY, UNCORRECTED_ERRORS, MISTAKES, RAW_WPM, WPM}
+
+		graph := g.PlotMany(
+			graphDatum,
+			g.SeriesColors(graphColors...),
+			g.SeriesLegends(graphLegends...),
+			g.Height(10),
+			g.Width(0), // auto scaling
+			g.LowerBound(0),
+			g.Precision(0),
+		)
+
+		fmt.Println(graph)
+
+	}
+
+	// print the reps table
+	table := tw.NewWriter(os.Stdout)
+	table.SetHeader(cols)
 	for _, rep := range reps {
 		repCols := []string{}
 		for _, c := range cols {
 			repCols = append(repCols, rep.ColumnString(c))
 		}
-		fmt.Printf("%s\n", strings.Join(repCols, "\t"))
+		table.Append(repCols)
 	}
-
+	table.Render()
```

Comment:
The reps table should start from most recent to least recent. This may be a job for the query itself so I'm not reversing the output myself all the time.

Also extract printing the reps table into its own function.